src/documentation/io/built-in-google-bigquery.md

```diff
@@ -0,0 +1,753 @@
+to write to the target table unless you specify a [create disposition]() of
+`CREATE_NEVER`. [Creating a table schema](#creating-a-table-schema) covers
+schemas in more detail.
+
+
+## Reading from BigQuery
+
+BigQueryIO allows you to read from a BigQuery table, or read specific fields
+using a query string. When you apply a BigQueryIO read transform in batch mode,
+Beam invokes a [BigQuery export request](https://cloud.google.com/bigquery/docs/exporting-data).
+Beam’s use of this API is subject to BigQuery's [Quota](https://cloud.google.com/bigquery/quota-policy#export)
+and [Pricing](https://cloud.google.com/bigquery/pricing) policies.
+
+
+<!-- Java specific -->
+
+{:.language-java}
+The Beam SDK for Java has two BigQueryIO read methods. Both of these methods
+allow you to read from a table, or read fields using a query string.
+
+{:.language-java}
+1. `readTableRows` returns a `PCollection` of BigQuery `TableRow`
+   objects.  Each element in the `PCollection` represents a single row in the
+   table. Integer values in the `TableRow` objects are encoded as strings to
+   match BigQuery's exported JSON format. This method is convenient, but can be
+   2-3 times slower in performance compared to `read(SerializableFunction)`. The
+   example code for reading from a table shows how to use `readTableRows`.
+
+2. `read(SerializableFunction)` reads Avro-formatted records and uses a
+   specified parsing function to parse them into a `PCollection` of custom typed
+   objects. Each element in the `PCollection` represents a single row in the
+   table. The example code for reading with a query string shows how to use
+   `read(SerializableFunction)`.
+
+{:.language-java}
+***Note:*** `BigQueryIO.read()` is deprecated as of Beam SDK 2.2.0. Instead, use
+`read(SerializableFunction<SchemaAndRecord, T>)` to parse BigQuery rows from
+Avro `GenericRecord` into your custom type, or use `readTableRows()` to parse
+them into JSON `TableRow` objects.
+
+
+<!-- Python specific -->
+
+{:.language-py}
+To read from a BigQuery table using the Beam SDK for Python, apply a `Read`
+transform on a `BigQuerySource`. Read returns a `PCollection` of dictionaries,
+where each element in the `PCollection` represents a single row in the table.
+Integer values in the `TableRow` objects are encoded as strings to match
+BigQuery's exported JSON format.
+
+
+### Reading from a table
+
+{:.language-java}
+To read an entire BigQuery table, use the `from` method with a BigQuery table
+name. This example uses `readTableRows`.
+
+{:.language-py}
+To read an entire BigQuery table, use the `table` parameter with the BigQuery
+table name.
+
+The following code reads an entire table that contains weather station data and
+then extracts the `max_temperature` column.
+
+```java
+{% github_sample /davidcavazos/beam/blob/bigquery-snippets/examples/java8/src/main/java/org/apache/beam/examples/snippets/Snippets.java tag:BigQueryReadTable
```

Comment:
This should probably use the snippet that uses read(SerializableFunction) - I think David has created those snippets already. Same for the other snippets.